### PR TITLE
Flesh out Reader and Writer APIs

### PR DESCRIPTION
**Pull request by joroKr21**
We are missing many combinators for those type classes.

#### Has the version number been increased?
 - [x] Yes!

### CODE REVIEW

File: src/main/scala/io/moia/protos/teleproto/Reader.scala [hunk 1]
```diff
@@ -43,18 +43,63 @@ trait Reader[-P, +M] { self =>
...
+  /**
+    * Transforms the protobuf before reading.
+    */
+  final def contramap[Q](f: Q => P): Reader[Q, M] =
+    protobuf => read(f(protobuf))
 
   /**
     * Transforms successfully read results with the option to fail.
     */
+  final def emap[N](f: M => PbResult[N]): Reader[P, N] =
```
Comment (aquamatthias) [inline src/main/scala/io/moia/protos/teleproto/Reader.scala:57]: Not sure I like the term emap. I think the design inspiration is coming from circe? 
There you pass in a function to Either - I assume the name is coming from that.
In analogy this should be called `pbmap`?

**Review by aquamatthias — APPROVED** [review]
One naming suggestion, otherwise LGTM.

File: src/main/scala/io/moia/protos/teleproto/Reader.scala (hunk 1 above)
Comment (joroKr21) [reply src/main/scala/io/moia/protos/teleproto/Reader.scala:57]: Agreed. I also want to add some tests but didn't find time today.

**Comment by joroKr21** [comment]
@aquamatthias it seems that we haven't automated releases. Actually we haven't even released 1.7.0

**Comment by aquamatthias** [comment]
Yes no automation here yet. Feel free to change.

**Pull request merged**